src/autopas/utils/MemoryProfiler.cpp

```diff
@@ -5,38 +5,28 @@
  */
 
 #include "MemoryProfiler.h"
-#include "Logger.h"
-
-#include <fcntl.h>
 #include <sys/stat.h>
-#include <unistd.h>
-#include <cstring>
+#include <fstream>
```

Comment:
```suggestion
#include <ifstream>
```
you are only using ifstream, so only include that.

Reply:
there is no `ifstream` header ^^